tla-io/src/main/scala/at/forsyte/apalache/tla/imp/SanyImporter.scala

```diff
@@ -16,37 +17,56 @@ import scala.util.{Failure, Success, Try}
    *
    * @param file
    *   an input file
-   * @param useRelativePathLookup
-   *   Whether to use [[file]]'s containing directory as library path. Disable this flag when you [[loadFromSource]]; we
-   *   then look up names in the paths given by the `TLA-Library` Java system variable instead.
    * @return
    *   the pair (the root module name, a map of modules)
    */
-  def loadFromFile(file: File, useRelativePathLookup: Boolean = true): (String, Map[String, TlaModule]) = {
+  def loadFromFile(file: File): (String, Map[String, TlaModule]) = {
     // create a string buffer to write SANY's error messages
     // use.toString() to retrieve the error messages
     val errBuf = new StringWriter()
 
-    // Resolver for filenames, patched for wired modules.
-    // If `file` has a parent directory and `useLibraryPaths` is set, look up files in this parent directory.
-    // Otherwise, look up files in the paths given by the `TLA-Library` Java system variable (used to wire in the
-    // Apalache Standard library in `src/tla` when running unit tests).
-    // Compatible with TLC at commit https://github.com/tlaplus/tlaplus/commit/b905a86e31714bf5002290141b51b41ffe280c8e
-    val filenameResolver = if (useRelativePathLookup) {
-      file.getAbsoluteFile().getParent() match {
-        case null          => SanyNameToStream()
-        case parentDirPath => SanyNameToStream(parentDirPath)
+    // Construct library lookup path:
+    //   1. Include the file's parent directory
+    val parentDirPath = file.getAbsoluteFile().getParent() match {
+      case null                  => Nil
+      case parentDirPath: String => Seq(parentDirPath)
+    }
+    //   2. Patch in Apache standard library.
+    //      When running Apalache from the fat JAR, these TLA+ modules are baked into the JAR.
+    //      Otherwise (e.g., running from `sbt`, an IDE, for unit tests), we include the standard library path for
+    //      lookup from the filesystem.
+    val standardLibraryPath = {
+      // Test if running from fat JAR; see also `build.sbt` and [[StandardLibrary.wiredModules]]
+      if (this.getClass().getClassLoader().getResource("tla2sany/StandardModules/Apalache.tla") != null) {
+        // Running from fat JAR with standard library baked in (in `build.sbt`).
+        // SANY looks up modules from the JAR as part of its base path, no need to adjust the lookup paths.
+        Nil
+      } else {
+        // Running from outside the fat JAR, we don't have a standard library bundled.
+        // Add $APALACHE_HOME/src/tla/ to lookup paths.
+        System.getenv("APALACHE_HOME") match {
+          // Warn if environment variable APALACHE_HOME is not set
+          case null =>
+            logger.warn("Not running from fat JAR and APALACHE_HOME is not set;")
+            logger.warn("will not be able to find the Apalache standard library.")
+            Nil
+          case apalacheHome: String => Seq(s"${apalacheHome}/src/tla")
+        }
```

Comment:
I really don't like the idea that we have two alternative module lookup strategies coded into the application logic in order to accommodate a particular development practice. I think it'd be much more preferable to expose the general purpose functionality we've discussed.

This looks to me like we're hardcoding a hack to inspect our own source code, and I don't understand why this is preferable to supporting general logic for looking up tla modules.

Reply:
We will always have two alternative module lookup strategies, no matter what. Fat JAR looks up from the baked-in files, anything else doesn't. Why would we introduce a way that potentially affects both, possibly shadowing the fat JAR lookup, if we can just fix the one that needs to do an extra lookup?

Reply:
I'm not opposed to having multiple module lookup strategies. I'm for it! What I'm opposed to is having one special module lookup strategy that only applies when the code is run in a pre-assembly, development state.

Reply:
I think we've hashed this out in the comments below.